scripts/tests/test_ci_workflow_permission_contract.py

```
@@ -0,0 +1,201 @@
+
+
+def _load_workflow(path: Path) -> dict:
+    return yaml.safe_load(path.read_text(encoding="utf-8")) or {}
+
+
+def _rank(level: str | None) -> int:
+    return _LEVEL_RANK.get(level or "none", 0)
+
+
+def _format(scopes: dict[str, str]) -> str:
+    return "{" + ", ".join(f"{s}: {l}" for s, l in sorted(scopes.items())) + "}"
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)